btk/create_observing_generator.py

```diff
@@ -35,69 +35,81 @@ def make_wcs(
     return w
 
 
-def default_obs_conditions(Args, band):
+def default_obs_conditions(survey_name, band):
     """Returns the default observing conditions from the WLD package
     for a given survey_name and band.
 
     Args:
-        Args: A `btk.config.SimulationParams` object containing parameters to generate blends
+        survey_name: Name of the survey which should be available in descwl
         band: filter name to get observing conditions for.
     Returns:
         `survey`: Dictionary containing the observing conditions and WCS information.
     """
     survey = descwl.survey.Survey.get_defaults(
-        survey_name=Args.survey_name, filter_band=band
+        survey_name=survey_name, filter_band=band
     )
     survey["center_sky"] = None
     survey["center_pix"] = None
     survey["projection"] = "TAN"
     return survey
 
 
-def generate(Args, obs_function=None):
-    """Generates class with observing conditions in each band.
+class ObservingGenerator:
+    def __init__(
+        self,
+        survey_name,
+        stamp_size,
+        obs_function=default_obs_conditions,
+        verbose=False,
+    ):
+        """Generates class with observing conditions in each band.
 
-    Args:
-        Args: Class containing input parameters.
-        obs_function: Function that outputs dict of observing conditions. If
-            not provided then the default `descwl.survey.Survey` values for the
-            corresponding Args.survey_name are used to create the
-            observing_generator.
+        Args:
+             obs_function: Function that outputs dict of observing conditions. If
+                           not provided, then the default `descwl.survey.Survey` values
+                           for the corresponding survey_name are used to create the
+                           observing_generator.
+             stamp_size: In arcseconds.
+        """
+        if survey_name not in btk.survey.surveys:
+            raise KeyError("Survey not implemented.")
+        self.survey_name = survey_name
+        self.bands = btk.survey.surveys["survey_name"]["bands"]
+        self.pixel_scale = btk.survey.surveys["survey_name"]["pixel_scale"]
+        self.stamp_size = stamp_size
+        self.obs_function = obs_function
```

Comment:
`obs_function` is currently a function as before, should it become a class like the sampling function ?

Reply:
yeah good point, I see two advantages with converting `obs_conditions` to a class, (1) code is more 'parallel' : obs_conditions are to observing generator as sampling function is to blend generator. (2) A class for obs_conditions would make more sense as it becomes more complicated e.g. for multi resolution.

Reply:
I'll go ahead and make this change